app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java

```diff
@@ -28,6 +28,10 @@
     Call<State> getFullProductByBarcode(@Path("barcode") String barcode,
                                         @Header("User-Agent") String header);
 
+    @GET("api/v0/product/{barcode}.json")
```

Comment:
api/v0/product/{barcode}.json?fields=ingredients_text_{lang},common_name_{lang},product_name_{lang}

example: https://world.openfoodfacts.org/api/v0/product/3302749508029.json.json?fields=ingredients_text_fr,generic_name_fr,product_name_fr

Reply:
okay

Reply:
java.lang.IllegalArgumentException: URL query string "fields=ingredients_text_{lang},common_name_{lang},product_name_{lang}}" must not have replace block. For dynamic query parameters use @Query.

Getting this error.

Reply:
or pass lang as @Path I guess.

Reply:
Yeah I did the same and got this error. 

@GET("api/v0/product/{barcode}.json?fields=ingredients_text_{lang},common_name_{lang},product_name_{lang}")
    Call<JsonNode> getFieldByLangCode(@Path("barcode") String barcode,
                                          @Path("lang") String lang);

Reply:
I think we cannot use {lang} in @get since field parameters can't be dynamic.

Reply:
@teolemon can you please suggest what changes are needed in the code?